### ISSUE TITLE
view audit trail with QA notes

### ISSUE THREAD

**Issue by kdionisio** [HumanExposure/factotum#1348]
As a Factotum product owner
I have a need to view the audit trail data related to composition document extraction together with QA notes
So that I can appropriately judge if an extraction script needs to be redone.

Both QA notes and audit trail logs need to be viewed together, as either one may contribute to a decision to redo an extraction script.

http://factotum.epa.gov/qa/compextractionscript/159/summary

Definition of Done:
On the QA summary page (/qa/compextractionscript/159/summary), view any stored audit trail data along with the QA notes
- all documents with QA notes should be included in the table, regardless of whether they are included in the QA group for that extraction script
 - audit trail data should be displayed alongside the QA notes summary data (e.g. table currently lists document name and then QA notes, add another column for audit trail data); audit trail data can be displayed as 'last edited at DD/XX etc', with hyperlink to popup that includes all information
- If the document does not include any QA notes but does include audit trail data, it should be included on the QA summary page, regardless of whether it is included in the QA group for that extraction script

**Comment by michael-barbour** [comment]
I've got a working version of what you're asking for.  Before I spent a lot of time polishing this I'd like to know if what I'm seeing sounds good and if you agree with some design choices I've had to make.

**TL;DR:**  The results in the audit log aren't going to be fully representative due to what data we collect.   If you're okay with that I'll proceed as planned.

**We don't "log" extracted text or data document changes.**
This isn't an issue but it bring into question what should be on the audit trail.  I've decided to implement this with the **combined AuditLogs for every chemical and functional use** attached to each extracted text.

**Every extracted use creates audit log rows on insert**
This means basically every result is going to have ~5 rows of edits from the creation of the chemical.  This could be useful, except I'm unsure what you're expecting to see.  If this were a data document from one of the groups with 4000 chemicals that could easily be 20,000 results just from the creation of the chemicals.

To combat this I'm **only showing updated rows** on the results.  Many of the results will be blank but you'll only see the things that had to be changed.

**Updated at**
The date for the updated at could be 1 of 3 different things.  It could be the date the document got updated last, the date the extracted text got updated last, or the date any one of the related chemicals got updated last.  I'm finding the most recent of the 3.

**Comment by kdionisio** [comment]
All of this sounds good to me, thank you! Agreed for sure that what I am looking for is the information on 'updated rows.' I do not need the information on initiation of the chemicals as you suspected.

**Comment by zachmunzenrider** [comment]
As Michael and I have been discussing amongst ourselves, this ticket is looking good, with one possible issue: I suspect that Kathie only wants documents to be listed in this datatable which have either or both 1) a QA Note, or 2) an indication in the auditlog that it has been updated since its creation. Michael has sketched out how that query would look, and we've Slacked Kathie to see if we're right about this
